src/test_lib_json/main.cpp

```diff
@@ -3286,11 +3286,11 @@ struct CharReaderAllowDropNullTest : JsonTest::TestCase {
     return [=](const Value& root) { JSONTEST_ASSERT_EQUAL(root, v); };
   }
 
-  ValueCheck objGetAnd(std::string idx, ValueCheck f) {
+  static ValueCheck objGetAnd(std::string idx, ValueCheck f) {
```

Comment:
Why do we add static here?

The fact that the body of the function HAPPENS to not use any nonstatic members of the enclosing class is an implementation detail. I would say these helpers reserve the right to modify the instance of the object on which they are called. clang-tidy outsmarting itself here again.